### PR TITLE
Fix deploymentconfig scale

### PR DESCRIPTION
**Pull request by liggitt**
3.6 and 3.7 shipped with the /apis/apps.openshift.io/deploymentconfigs/scale subresource returning apps.openshift.io/v1 Scale objects

This is a non-standard Scale object that the HPA will never be able to make use of.

The intent was to continue send the same thing as /oapi/v1/deploymentconfigs/scale: extensions/v1beta1 Scale objects

This PR fixes the groupified API to send/receive the correct type.

It also updates the internal UpdateScale() client method to send extensions/v1beta1 Scale objects

### CODE REVIEW

File: pkg/apps/apiserver/apiserver.go [hunk 1]
```diff
@@ -89,6 +90,12 @@ func (c completedConfig) New(delegationTarget genericapiserver.DelegationTarget)
 	apiGroupInfo := genericapiserver.NewDefaultAPIGroupInfo(appsapiv1.GroupName, c.ExtraConfig.Registry, c.ExtraConfig.Scheme, parameterCodec, c.ExtraConfig.Codecs)
 	apiGroupInfo.GroupMeta.GroupVersion = appsapiv1.SchemeGroupVersion
 	apiGroupInfo.VersionedResourcesStorageMap[appsapiv1.SchemeGroupVersion.Version] = v1Storage
+
+	if apiGroupInfo.SubresourceGroupVersionKind == nil {
+		apiGroupInfo.SubresourceGroupVersionKind = map[string]schema.GroupVersionKind{}
+	}
+	apiGroupInfo.SubresourceGroupVersionKind["deploymentconfigs/scale"] = v1beta1extensions.SchemeGroupVersion.WithKind("Scale")
```
Comment (soltysh) [inline pkg/apps/apiserver/apiserver.go]: Hmm... is it worth adding it here? The rebase (#17503) is going to remove `SubresourceGroupVersionKind` or are we not doing the rebase?

**Comment by liggitt** [comment]
opened 3.9 pull against master at https://github.com/openshift/origin/pull/17587

**Comment by liggitt** [comment]
extended_conformance_install has passed three times and failed four times with unrelated failures on this commit. this is blocking the 1.8.4 rebase... would like to green button it

**Comment by deads2k** [comment]
> extended_conformance_install has passed three times and failed four times with unrelated failures on this commit. this is blocking the 1.8.4 rebase... would like to green button it

I better not turn purple this time...

**Pull request merged**